### PR TITLE
request: add LocalIP

### PR DESCRIPTION
**Pull request by miekg**
Fix TODO that was added: Add LocalIP and test the Clear() method.

Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

File: request/request_test.go [hunk 1]
```diff
@@ -202,3 +202,22 @@ func testRequest() Request {
...
+	if p == "" {
+		t.Fatalf("Failed to get Port from request")
+	}
+	st.Clear()
+	if st.ip != "" {
+		t.Errorf("Expected st.ip to be cleared after Clear")
+	}
+
+	if st.port != "" {
+		t.Errorf("Expected st.ip to be cleared after Clear")
```
Comment (Isolus) [inline request/request_test.go]: I think this should be "Expected st.port..."?

Comment (miekg) [reply request/request_test.go]: yep!

**Comment by miekg** [comment]
moved to Errorf for the whole file as well. ptal

**Review by Isolus — APPROVED** [review]
LGTM

**Pull request merged**